ipa-core/src/helpers/transport/in_memory/config.rs

```diff
@@ -93,7 +109,10 @@ pub struct MaliciousHelper<F> {
     inner: F,
 }
 
-impl<F: Fn(&MaliciousHelperContext, &mut Vec<u8>) + Send + Sync> MaliciousHelper<F> {
+impl<F> MaliciousHelper<F>
+where
+    F: Fn(MaliciousHelperContext, &mut Vec<u8>) -> Pin<Box<dyn Future<Output = ()> + Send>>,
```

Comment:
Do we need this given that it is only used in non-async context? it may be better to convert non-async code to async inside this helper - non-async code is much easier to deal with